src/components/3d/SimpleModelView.tsx

```diff
@@ -0,0 +1,148 @@
+import * as React from 'react';
+import {
+  Animated,
+  PanResponder,
+  StyleSheet,
+  View,
+  ViewStyle,
+} from 'react-native';
+import { WebView } from 'react-native-webview';
+import styled from 'styled-components';
+import { ImgixImage } from '@rainbow-me/images';
+import { padding, position } from '@rainbow-me/styles';
+
+export type ModelViewerProps = {
+  readonly setLoading: (loading: boolean) => void;
+  readonly loading: boolean;
+  readonly style?: ViewStyle;
+  readonly uri?: string;
+  readonly alt?: string;
+  readonly fallbackUri?: string;
+};
+
+const StyledWebView = styled(WebView)`
+  ${position.size('100%')};
+`;
+
+const ProgressIndicatorContainer = styled(Animated.View)`
+  ${position.size('100%')};
+  position: absolute;
+  align-items: flex-end;
+  justify-content: flex-end;
+  ${padding(10)};
+`;
+
+export default function ModelViewer({
+  loading,
+  setLoading,
+  style,
+  uri,
+  alt,
+  fallbackUri,
+}: ModelViewerProps): JSX.Element {
+  const opacity = React.useMemo(() => new Animated.Value(1), []);
+  const [visibility, setVisibility] = React.useState<boolean>(false);
+  const [progress, setProgress] = React.useState<number>(0);
+  const originWhiteList = React.useMemo(() => ['*'], []);
+  const onMessage = React.useCallback(
+    ({ nativeEvent: { data } }) => {
+      const { type, payload } = JSON.parse(data);
+      if (type === 'progress') {
+        setProgress(payload as number);
+      } else if (type === 'model-visibility') {
+        setVisibility(payload as boolean);
+      }
+    },
+    [setProgress, setVisibility]
+  );
+  const fallbackSource = React.useMemo(
+    () => ({
+      uri: fallbackUri,
+    }),
+    [fallbackUri]
+  );
+  const source = React.useMemo(
```

Comment:
I think this template can we move this to the constant... or even another file. I think it doesn't look cool, cause it cannot follow any reasonable formatting.